web/modules/custom/server_general/src/ThemeTrait/PersonCardThemeTrait.php

```diff
@@ -0,0 +1,143 @@
+use Drupal\server_general\ThemeTrait\Enum\FontSizeEnum;
+use Drupal\server_general\ThemeTrait\Enum\FontWeightEnum;
+use Drupal\server_general\ThemeTrait\Enum\TextColorEnum;
+use Drupal\server_general\ThemeTrait\Enum\BackgroundColorEnum;
+
+/**
+ * Helper for a single Person Card.
+ */
+trait PersonCardThemeTrait {
+  use ElementWrapThemeTrait;
+  use InnerElementLayoutThemeTrait;
+
+  /**
+   * Wraps a render array with a background color.
+   */
+  protected function wrapBackgroundColor(array $element, BackgroundColorEnum $color): array {
+    $class = match($color) {
+      BackgroundColorEnum::LightGray => 'bg-gray-200',
+      BackgroundColorEnum::Transparent => 'bg-transparent',
+      BackgroundColorEnum::LightGreen => 'bg-green-100',
+    };
+
+    return [
+      '#type' => 'container',
+      '#attributes' => ['class' => [$class, 'rounded-lg', 'px-2', 'py-1', 'inline-block']],
+      'content' => $element,
+    ];
+  }
+
+  /**
+   * Build a single Person card.
+   */
+  protected function buildElementPersonCard(
+    string $image_url,
+    string $alt,
+    string $name,
+    string $detail,
+    string $badge,
+    string $email,
+    string $phone
+  ): array {
+    // Main content (image, name, detail, badge).
+    $main_inner = [];
+
+    // Image.
+    $main_inner[] = $this->wrapRoundedCornersFull([
+      '#theme' => 'image',
+      '#uri' => $image_url,
+      '#alt' => $alt,
+      '#width' => 128,
+    ]);
+
+    // Name (bold).
+    $name_el = $this->wrapTextFontWeight($name, FontWeightEnum::Bold);
+    $main_inner[] = $this->wrapTextCenter($name_el);
+
+    // Detail (sm, gray).
+    $detail_el = $this->wrapTextResponsiveFontSize($detail, FontSizeEnum::Sm);
+    $detail_el = $this->wrapTextCenter($detail_el);
+    $main_inner[] = $this->wrapTextColor($detail_el, TextColorEnum::Gray);
+
+    // Badge (xs)
+    $badge_el = $this->wrapTextResponsiveFontSize($badge, FontSizeEnum::Xs);
+    $badge_el = $this->wrapTextCenter($badge_el);
+    $badge_el = $this->wrapTextColor($badge_el, TextColorEnum::DarkGreen);
+    $badge_el = $this->wrapBackgroundColor($badge_el, BackgroundColorEnum::LightGreen);
+    $main_inner[] = $badge_el;
+
+    // Email and phone row, each wrapped in a div with icon.
+    $email_row = [
+      '#type' => 'container',
+      '#attributes' => ['class' => ['flex', 'items-center', 'gap-2', 'pt-4', 'pb-4']],
+      'icon' => [
+        '#markup' => '<i class="fa fa-envelope text-gray-400"></i>',
+      ],
+      'link' => [
+        '#type' => 'link',
+        '#title' => $this->wrapTextColor(
+          $this->wrapTextResponsiveFontSize('Email', FontSizeEnum::Sm),
+          TextColorEnum::DarkGray
+        ),
+        '#url' => Url::fromUri('mailto:' . $email),
+      ],
+    ];
+
+    // Divider span.
+    $divider = [
+      '#markup' => '<span class="mx-4 self-stretch border-r border-gray-200"></span>',
+    ];
+
+    // Phone row.
+    $phone_row = [
```

Comment:
This is still foreign to the code base. We don't have stuff like that on Drupal-starter, we instead hand it over to twig files